src/test/java/io/dropwizard/sharding/dao/locktest/LockTest.java

```diff
@@ -254,4 +255,288 @@ public void testPersist_alreadyExistingSame() throws Exception {
+                .value("Hello1")
+                .build()).get();
+
+        final SomeOtherObject c2 = relationDao.save(p1.getMyId(), SomeOtherObject.builder()
+                .my_id(p1.getMyId())
+                .value("Hello2")
+                .build()).get();
+
+        final String p2Id = "1";
+        final SomeLookupObject p2 = SomeLookupObject.builder()
+                .myId(p2Id)
+                .name("Parent 2")
+                .build();
+
+        final SomeOtherObject c3 = relationDao.save(p2.getMyId(), SomeOtherObject.builder()
+                .my_id(p2.getMyId())
+                .value("Hello3")
+                .build()).get();
+
+        lookupDao.save(p1);
+        lookupDao.save(p2);
+
+        //test full update
+        final DetachedCriteria criteria = DetachedCriteria.forClass(SomeOtherObject.class)
+                .add(Restrictions.eq("my_id", p1.getMyId()));
+
+        final String childModifiedValue = "Hello Modified";
+        final String parentModifiedValue = "Parent Changed";
+
+        lookupDao.lockAndGetExecutor(p1.getMyId())
+                .scrollAndUpdate(relationDao, criteria,ScrollMode.FORWARD_ONLY, scrollableResults -> {
+                    final List<SomeOtherObject> modified = new ArrayList<>();
+                    while(scrollableResults.next()) {
+                        final SomeOtherObject entity = (SomeOtherObject) scrollableResults.get(0);
+                        entity.setValue(childModifiedValue);
+                        modified.add(entity);
+                    }
+                    return modified;
+                })
+                .mutate(parent -> parent.setName(parentModifiedValue))
+                .execute();
+
+        Assert.assertEquals(childModifiedValue,relationDao.get(p1.getMyId(), c1.getId()).get().getValue());
+        Assert.assertEquals(childModifiedValue,relationDao.get(p1.getMyId(), c2.getId()).get().getValue());
+        Assert.assertEquals(parentModifiedValue,lookupDao.get(p1Id).get().getName());
+        Assert.assertEquals(p1Id,lookupDao.get(p1Id).get().getMyId());
+
+        Assert.assertEquals("Hello3",relationDao.get(p2.getMyId(), c3.getId()).get().getValue());
+        Assert.assertEquals(p2Id,lookupDao.get(p2Id).get().getMyId());
+        Assert.assertEquals("Parent 2",lookupDao.get(p2Id).get().getName());
+
+
+
+        //test partial update
+        final String childModifiedValue2 = "Hello Modified Partial";
+        final String parentModifiedValue2 = "Parent Changed Partial";
+        lookupDao.lockAndGetExecutor(p1.getMyId())
+                .scrollAndUpdate(relationDao, criteria, ScrollMode.FORWARD_ONLY, scrollableResults -> {
+                    final List<SomeOtherObject> modified = new ArrayList<>();
+
+                    while(scrollableResults.next()) {
+                        final SomeOtherObject entity = (SomeOtherObject) scrollableResults.get(0);
+
+                        if (entity.getId() == c1.getId()) {
```

Comment:
This is not proper code. Basically, read the above comment. If a single object needs to be updated, only that one needs to be selected by the criteria. For modifying other objects, .update(...) methods can be chained accordingly.